pkg/route/graph/analysis/analysis.go

```diff
@@ -86,3 +90,25 @@ func FindMissingTLSTerminationType(g osgraph.Graph, f osgraph.Namer) []osgraph.M
 
 	return markers
 }
+
+func FindPathBasedPassthroughRoutes(g osgraph.Graph, f osgraph.Namer) []osgraph.Marker {
+	markers := []osgraph.Marker{}
+
+	for _, uncastRouteNode := range g.NodesByKind(routegraph.RouteNodeKind) {
+		routeNode := uncastRouteNode.(*routegraph.RouteNode)
+
+		if len(routeNode.Spec.Path) > 0 && routeNode.Spec.TLS != nil && routeNode.Spec.TLS.Termination == routeapi.TLSTerminationPassthrough {
+			markers = append(markers, osgraph.Marker{
+				Node: routeNode,
+
+				Severity: osgraph.ErrorSeverity,
+				Key:      PathBasedPassthroughErr,
+				Message:  fmt.Sprintf("%s is path-based and uses passthrough termination, which is an invalid combination.", f.ResourceName(routeNode)),
+				// TODO: Provide the commands separately for copy/paste
+				Suggestion: osgraph.Suggestion(fmt.Sprintf("Either remove the path or switch the route to use edge termination")),
```

Comment:
Based on the already existing validation (prior to merging https://github.com/openshift/origin/pull/6037), 1. reencrypt requires a destination CA cert file and 2. passthrough doesn't accept any certificates at all. So in order to switch to reencrypt the user would have to add that certificate additionally to changing the termination type. Instead switching to edge takes just the termination type. I don't feel strongly, I can add the reencrypt suggestion as well if you want.